### PR TITLE
Fix/on permission change after app permission consent

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Fixes #https://github.com/smartdevicelink/sdl_core/issues/2072
This PR is **not ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Tested with https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/test_scripts/Policies/External_UCS/External_Consent_OFF/001_ATF_Policies_External_Consent_OFF_disallowed_by_external_consent_entities_on_data_consent_disallowed.lua 

### Summary
Process consent update even if not groups specified in OnAppPermissionConsent
Add RC groups to permission groups on Permissions check. 

### CLA
- [ x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by LuxoftAKutsan** [comment]
Depends in merge #2085

File: src/components/policy/policy_external/src/policy_manager_impl.cc [hunk 1]
```diff
@@ -588,13 +588,15 @@ void PolicyManagerImpl::CheckPermissions(const PTString& app_id,
...
-    const policy_table::Strings app_groups = access_remote_->GetGroups(who);
-#else   // SDL_REMOTE_CONTROL
-    const policy_table::Strings app_groups =
+    policy_table::Strings app_groups =
         GetGroupsNames(app_group_permissions);
-#endif  // SDL_REMOTE_CONTROL
+
+//#ifdef SDL_REMOTE_CONTROL
+//    ApplicationOnDevice who = {device_id, app_id};
+//    const auto rc_groups = access_remote_->GetGroups(who);
```
Comment (JackLivio) [inline src/components/policy/policy_external/src/policy_manager_impl.cc]: I think this code should just be removed. Any reason for keeping it commented out?

File: src/components/policy/policy_external/src/policy_manager_impl.cc [hunk 2]
```diff
@@ -615,7 +610,8 @@ void PolicyManagerImpl::CheckPermissions(const PTString& app_id,
   }
 
   const bool known_rpc = rpc_permissions.end() != rpc_permissions.find(rpc);
-  LOG4CXX_INFO(logger_, "Is known rpc " << known_rpc);
+  LOG4CXX_DEBUG(logger_, "Is known rpc " <<
+                (known_rpc ? "true" : "false") );
```
Comment (AByzhynar) [inline src/components/policy/policy_external/src/policy_manager_impl.cc]: @LuxoftAKutsan :+1:

File: src/components/policy/policy_external/src/policy_manager_impl.cc [hunk 3]
```diff
@@ -986,10 +984,6 @@ void PolicyManagerImpl::SetUserConsentForApp(
     const PermissionConsent& permissions, const NotificationMode mode) {
   LOG4CXX_AUTO_TRACE(logger_);
 
-  if (permissions.group_permissions.empty()) {
-    LOG4CXX_DEBUG(logger_, "Permissions list is empty, skipping update.");
-    return;
-  }
```
Comment (AByzhynar) [inline src/components/policy/policy_external/src/policy_manager_impl.cc]: @LuxoftAKutsan Why this code was removed?

Comment (LuxoftAKutsan) [reply src/components/policy/policy_external/src/policy_manager_impl.cc]: We should be able to apply user consent even is groups are empty

**Pull request merged**